### PR TITLE
Add a github service with issue/PR expansion

### PR DESCRIPTION
**Pull request by Kegsay**
This adds a Github service which will expand issues that look like `owner/repo#11`. It uses an unauthenticated client for now (so no database required!), and is mainly so I can get to grips with the Go-NEB plugin architecture.

### CODE REVIEW

File: src/github.com/matrix-org/go-neb/services/github/github.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+func (s *githubService) ServiceUserID() string { return s.UserID }
+func (s *githubService) ServiceID() string     { return s.id }
+func (s *githubService) ServiceType() string   { return "github" }
+func (s *githubService) RoomIDs() []string     { return s.Rooms }
+func (s *githubService) Plugin(roomID string) plugin.Plugin {
+	return plugin.Plugin{
+		Commands: []plugin.Command{},
+		Expansions: []plugin.Expansion{
+			plugin.Expansion{
+				Regexp: regexp.MustCompile(ownerRepoIssueRegex),
```
Comment (NegativeMjark) [inline src/github.com/matrix-org/go-neb/services/github/github.go]: I'd prefer to have the regexp stored as a top-level variable, That way the MustCompile will fail earlier if the rexexp is bad. (Also means that we know that we will only construct the regexp once.)

File: src/github.com/matrix-org/go-neb/services/github/github.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	httpCli := oauth2.NewClient(oauth2.NoContext, tokenSource)
+	return github.NewClient(httpCli)
+}
+
+// ownerRepoNumberFromText parses a GH issue string that looks like 'owner/repo#11'
+// into its constituient parts. Returns: owner, repo, issue#.
+func ownerRepoNumberFromText(ownerRepoNumberText string) (string, string, int, error) {
+	// TODO: cache this?
+	re := regexp.MustCompile(ownerRepoIssueRegex)
```
Comment (NegativeMjark) [inline src/github.com/matrix-org/go-neb/services/github/github.go]: Again, I'd prefer it if the regexp was stored at the top-level.

Comment (Kegsay) [inline src/github.com/matrix-org/go-neb/services/github/github.go]: done

File: src/github.com/matrix-org/go-neb/services/github/github.go (hunk 1 above)
Comment (Kegsay) [inline src/github.com/matrix-org/go-neb/services/github/github.go]: done

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**